taichi/backends/metal/struct_metal.cpp

```diff
@@ -227,8 +227,11 @@ class StructCompiler {
       emit("  SNodeRep_{} rep_;", snty_name);
       emit("}};");
     } else {
-      TI_ERROR("SNodeType={} not supported on Metal", snode_type_name(snty));
-      TI_NOT_IMPLEMENTED;
+      TI_ERROR(
+          "SNodeType={} not supported on OpenGL\n"
+          "Consider use ti.init(ti.cpu) or ti.init(ti.cuda) if you "
```

Comment:
Unfortunately, Mac doesn't have CUDA.
```suggestion
          "Consider using `bitmasked` if you want `pointer`, otherwise please use arch=ti.cpu"
```